main.go

```diff
@@ -40,10 +60,11 @@ func genDownloadFileChannel(in <-chan string) <-chan structs.DownloadFile {
 			req.Header.Set("User-Agent", "GoLang Img Downloadeder/0.1")
 
 			resp, err := client.Do(req)
-			defer resp.Body.Close()
 			if err != nil {
 				log.Fatal(err)
 			}
+			// Not that it really matters in this case because of the log.Fatal
+			defer resp.Body.Close()
```

Comment:
Also I'm not 100% sure this is correct for http response bodies, but I've had to use a similar pattern with database result cursors.